apsuite/commissioning_scripts/measure_idffwd.py

```diff
@@ -200,15 +196,31 @@ def _init_ffwd_table(self):
 
     def _static_get_trajectory(self):
         """."""
+        sofb = self.devices['sofb']
+
         # reset SOFB buffer and wait for filling
-        self.devices['sofb'].cmd_reset()
-        _time.sleep(2 * self.params.sofb_nrpts * 1/MeasAPUFFWD._SOFB_FREQ)
+        sofb.cmd_reset()
+        wait_factor = 1 + self.params.sofb_overwait/100
+        wait_nominal = self.params.sofb_nrpts * 1/MeasAPUFFWD._SOFB_FREQ
+        _time.sleep(wait_factor * wait_nominal)
+
+        # get trajectory
+        traj = _np.vstack((sofb.trajx, sofb.trajy))
 
-        trajx = self.devices['sofb'].trajx
-        trajy = self.devices['sofb'].trajy
-        traj = _np.vstack((trajx, trajy))
         return traj
 
-    def _print(self, message):
+    def _static_move(self, phase):
+        """."""
+        self._print('- moving to phase {} ... '.format(phase), end='')
+        self.devices['apu'].phase_speed = self.params.phase_speed
+        self.devices['apu'].phase = phase
+        self.devices['apu'].cmd_move()
+        _time.sleep(MeasAPUFFWD._MOVE_SLEEP)
+        while self.devices['apu'].is_moving:
```

Comment:
I think this wait for move to end routine should be implemented in the APU class.